pkg/restapi/dochandler/resolvehandler.go

```diff
@@ -60,6 +60,9 @@ func (o *ResolveHandler) doResolve(id string) (*document.ResolutionResult, error
 
 	doc, err := o.resolver.ResolveDocument(id)
 	if err != nil {
+		if strings.Contains(err.Error(), "bad request") {
```

Comment:
Can we instead define a 'BadRequest' error and then use errors.Is(...) to determine that it's the cause? I find that matching a string within the error is not reliable. I was already thinking about refactoring the code that searches for "not found".

Reply:
Should I provide details about error. We usually use this pattern if error is constant e.g. 
var ErrNotFound = errors.New("did not found under given key")

Reply:
I think that a new 'BadRequestError' type should be defined which has a code. The code contains a specific value that's defined in an enumeration. This is what's done for the cas REST endpoint: https://github.com/trustbloc/sidetree-fabric/blob/master/pkg/rest/dcashandler/retrieveerror.go

Reply:
You have 5 status codes there. If I would capture all validations it would probably be more than 30 status codes plus they are all over the place because of the nature of resolution with initial state (parse did, parse initial state parameter, validate initial document, parse operation which involves suffix, delta patches (keys, services), reveal values, commitment values, hashes etc). I can create new issue for that and we can address it later on.

Reply:
OK, this can be done later.

Reply:
Created #264